### PR TITLE
[W08H02] added behavioral and structural tests

### PR DESCRIPTION
**Pull request by TheAleWu**
Added structural behavioral tests which include the Artemis tests and examples aswell as own edge cases

### CODE REVIEW

**Comment by MaximilianAnzinger** [comment]
@Mauri2070 can you take look at the structural tests?
Originally the README states

> To ensure the solution is not getting revealed, the tests should be Input/Output-Tests only.

However, we've deviated from this statement in some exercises. Since this is yours I suggest you should decide.
As far as I've seen the structural tests only test methods/attributes that are required by the assignment anyway so I personally don't see why this should be a problem.

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by Mauri2070 — APPROVED** [review]
Structural tests don't reveal anything not explicitly mentioned in the task description. They can be merged.

File: w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package pgdp.pools.structural;
+
+import java.lang.reflect.Modifier;
+import java.util.function.Function;
```
Comment (nilsreichardt) [inline w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java:4]:
```suggestion
```

This import is unused, right?

File: w08h02/test/pgdp/pools/TaskPoolBehaviorTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package pgdp.pools;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
```
Comment (nilsreichardt) [inline w08h02/test/pgdp/pools/TaskPoolBehaviorTest.java:4]:
```suggestion
```

This import is unused, right?

**Review by nilsreichardt — CHANGES_REQUESTED** [review]
Thanks for the tests! 🎉

Have you used auto-format for the code? I'm using VS Code and I could format some files. It would be great to have all files formatted so that when someone else makes a new change and uses auto-format not the whole file is formatted.

Otherwise, LGTM 👍

File: w08h02/test/pgdp/pools/TaskPoolBehaviorTest.java (hunk 2 above)
Comment (TheAleWu) [reply w08h02/test/pgdp/pools/TaskPoolBehaviorTest.java:4]: Yep, you're right. I've removed it in the next commit 👍

File: w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java (hunk 1 above)
Comment (TheAleWu) [reply w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java:4]: Nope, Function is used in line 20 and 26 ^^

**Comment by TheAleWu** [comment]
> Thanks for the tests! 🎉
> 
> Have you used auto-format for the code? I'm using VS Code and I could format some files. It would be great to have all files formatted so that when someone else makes a new change and uses auto-format not the whole file is formatted.
> 
> Otherwise, LGTM 👍

Thanks for the review! 🙏 
And yep, i've auto-formatted the code, but I used IntelliJ's formatting (and additionally a custom Code Style) which might deviate from VS Codes.

File: w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java (hunk 1 above)
Comment (nilsreichardt) [reply w08h02/test/pgdp/pools/structural/TaskFunctionStructuralTest.java:4]: Ah wrong file, sorry

File: w08h02/test/pgdp/pools/structural/TaskStructuralTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package pgdp.pools.structural;
+
+import java.lang.reflect.Modifier;
+import java.util.function.Function;
```
Comment (nilsreichardt) [inline w08h02/test/pgdp/pools/structural/TaskStructuralTest.java:4]:
```suggestion
```

This is the line that is unused 😅
